dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,287 @@
+package com.datadog.gradle.plugin
+
+import java.io.File
+import java.util.stream.Stream
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+
+internal class DdAndroidGradlePluginFunctionalTest {
+    @TempDir
+    lateinit var testProjectDir: File
+    lateinit var mainSrcDir: File
+    lateinit var settingsFile: File
+    lateinit var buildFile: File
+    lateinit var manifestFile: File
+    lateinit var gradlePropertiesFile: File
+
+    @BeforeEach
+    fun `set up`() {
+        mainSrcDir = File(testProjectDir, "src/main").apply { mkdirs() }
+        settingsFile = File(testProjectDir, "settings.gradle")
+        buildFile = File(testProjectDir, "build.gradle")
+        gradlePropertiesFile = File(testProjectDir, "gradle.properties")
+        manifestFile = File(mainSrcDir, "AndroidManifest.xml")
+        stubFile(settingsFile, SETTINGS_FILE_CONTENT)
+        stubFile(manifestFile, ANDROID_MANIFEST_FILE_CONTENT)
+        stubFile(gradlePropertiesFile, GRADLE_PROPERTIES_FILE_CONTENT)
+    }
+
+    // region Assemble
+
+    @Test
+    fun `M success W assemble`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { build cache }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--build-cache", "assemble")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assemble { configuration cache, checkProjectDependecies enabled }`() {
```

Comment:
Why should the plugin fail in this case? Also can we be a bit more precise in what kind of exception we're expecting ?

Reply:
Good catch, this one should not fail actually is just that I need to test the `assembleDebug` instead as `assemble` contains `assembleRelease` also and that ones fails for other reasons than our code (see the comment below).

Reply:
The reason why this test should fail is that our `checkDepsTask` is not aligned with the `--configuration-cache` requirements because we are using the `project` inside. @0xnm tried to fix this a while ago but it seems too tricky for now. We have an entry in our `Readme` file that states this but I will also add an explanation into this test.